com.thelocalmarketplace.software/src/com/thelocalmarketplace/software/Session.java

```diff
@@ -329,6 +349,19 @@ private void resume() {
 	}
 
 	/**
+	 * Enters the adding membership mode for the customer.
+	 *
+	 * @throws InvalidActionException
+	 */
+	public void enteringMembership() {
+		if (sessionState == SessionState.IN_SESSION) {
+			membership.setAddingItems(true);
+			} else {
+				throw new InvalidActionException("Cannot enter membership if session is not in adding items state");
+			}
+	}
+
+
 	 * Places a previously disabled session into the PRE_SESSION state
```

Comment:
Careful, the first /** is getting removed which causes the comment to be interperted as code.